Automattic/vip-cli
# PLAT-1575: Adding @automattic/vip-search-replace into the public cli

## Description

This PR aims to add search and replace functionality to two places in the public CLI.

1. In a stand-alone command that can be used to run s&r without performing a DB import
1. As part of the SQL import command, so that s&r can occur before the import is executed

Note: you can pass an `--in-place` option to either command to have the search and replace utility overwrite the original input file.  When not provided and input of `filename.ext` creates an output file of `filename.out.ext`.

## Steps to Test

**Example to test s&r alone:**

1. Check out PR.
1. `npm i`
1. Get a SQL file to run s&r on (eg. `scp vipdev@go:/home/vipdev/Git/vip-go-internal-cli/test-dumpl.sql ./`)
1. Run `npm run build && VIP_PROXY="" API_HOST=http://localhost:4000 node ./dist/bin/vip-search-replace test-dumpl.sql --search-replace=countrycode.com,c0untryc0de.com --search-replace=thing1,thing2`
1. Verify that the original file created is present and contains the original SQL details with the s&r completed

**Example to test s&r with an import:**

1. Check out PR
1. `npm i`
1. Get a SQL file that you want to upload
1. *** Setup your local to use the Import flow (need to add details here)
1. Run `npm run build && VIP_PROXY="" API_HOST=http://localhost:4000 node ./dist/bin/vip-import-sql test-dumpl.sql --search-replace=countrycode.com,c0untryc0de.com`
1. See that the import succeeded and that the s&r operated correctly before the import



---
_This PR was started by: [git wf pr](https://github.com/groupon/git-workflow/releases/tag/v1.1.0)_
